plugin/src/main/kotlin/com/varabyte/kobweb/intellij/util/kobweb/KobwebUtils.kt

```diff
@@ -24,7 +44,7 @@ private fun PsiElement.isInKobwebSource(): Boolean {
  *   By default, this is false, so unless this is explicitly set, this extension point will be active inside Kobweb
  *   source.
  */
-fun PsiElement.isInKobwebReadContext(limitTo: Set<KobwebProject.Type> = KobwebProject.Types.Framework, excludeKobwebSource: Boolean = false): Boolean {
+fun PsiElement.isInKobwebReadContext(limitTo: Set<KobwebProjectType> = KobwebProjectTypes.Framework, excludeKobwebSource: Boolean = true): Boolean {
     return isInKobwebContext { it.type in limitTo } || (!excludeKobwebSource && isInKobwebSource())
```

Comment:
Not sure I follow. Here's all cases:

**A Kobweb module, excludeKobwebSource == true**
* isInKobwebContext == true
* isInKobwebSource == false
return true || (false && false) --> true :heavy_check_mark: 

**A Kobweb module, excludeKobwebSource == false**
* isInKobwebContext == true
* isInKobwebSource == false
return true || (true && false) --> true :heavy_check_mark: 

**A non Kobweb module, excludeKobwebSource == true**
* is InKobwebContext == false
* isInKobwebSource == false
return false || (false && false) --> false :heavy_check_mark: 

**A non Kobweb module, excludeKobwebSource == false**
* is InKobwebContext == false
* isInKobwebSource == false
return false || (true && false) --> false :heavy_check_mark: 

**A Kobweb library, excludeKobwebSource == true**
* isInKobwebContext == true
* isInKobwebSource == false
return true || (false && false) --> false :heavy_check_mark: 

**A Kobweb library, excludeKobwebSource == false**
* isInKobwebContext == false
* isInKobwebSource == false
return false || (true && false) --> false :heavy_check_mark: 

**A Kobweb source file, e.g. Colors.kt, excludeKobwebSource == true**
* isInKobwebContext == false
* isInKobwebSource == true
return false || (false && true) --> false :heavy_check_mark: 

**A Kobweb source file, e.g. Colors.kt, excludeKobwebSource == false**
* isInKobwebContext == false
* isInKobwebSource == true
return false || (true && true) --> true :heavy_check_mark: 

---

I'm guessing you were making a different point through. Apologies if I missed it.

Reply:
Some of these examples are incorrect, though thats besides the point. What I was wondering is why this toggle only disables returning true for the official kobweb libraries. I think you implemented this, because you wanted usages of this function, to be able to ignore contexts of non-local files. I dont see why one would ever want to ignore official kobweb libraries, but not unofficial ones?

Reply:
Which examples did I screw up?

Writing a Kobweb application is different than writing Kobweb framework code itself. (You'll never define a `@Page` inside the Kobweb framework, for example).

My expectation is that most inspections we write won't apply to the Kobweb framework code, so it seems to make sense to ask developers (i.e. our future selves) to consider this each time. 

I can't predict the future -- if it turns out that most of the time we just keep setting `excludeKobwebSources` to false, then I should probably just remove the parameter at that time.

Another approach is to do this, for example, in getColor, if you prefer:

```kotlin
!element.isInReadOnlyKobwebContext() || !element.isInKobwebSources() -> null
```

Do you like that better?

Reply:
Resolved off-github